test/common/tcp_proxy/tcp_proxy_test.cc

```diff
@@ -735,6 +751,47 @@ TEST_P(TcpProxyTest, UpstreamDisconnectDownstreamFlowControl) {
   filter_callbacks_.connection_.runLowWatermarkCallbacks();
 }
 
+TEST_P(TcpProxyTest, ReceiveBeforeConnectBuffersOnEarlyData) {
```

Comment:
Please add a test variant that sends an end_data with the early data.

Reply:
Done